armotypes/cachetypes.go

```diff
@@ -1,64 +1,26 @@
 package armotypes
 
 import (
-	"fmt"
-	"strings"
 	"time"
 )
 
 type DataType string
 
 // PortalCache is an auxiliary structure to store cache data
-type PortalCache struct {
-	GUID         string      `json:"guid" bson:"guid"`
-	Name         string      `json:"name,omitempty" bson:"name,omitempty"`
-	DataType     DataType    `json:"dataType,omitempty" bson:"dataType,omitempty"`
-	Data         interface{} `json:"data,omitempty" bson:"data,omitempty"`
-	CreationTime string      `json:"creationTime" bson:"creationTime"`
-	UpdatedTime  string      `json:"lastUpdated,omitempty" bson:"lastUpdated,omitempty"`
-	ExpiryTime   time.Time   `json:"expiryTime,omitempty" bson:"expiryTime,omitempty"`
+type PortalCache[T any] struct {
+	GUID         string    `json:"guid" bson:"guid"`
+	Name         string    `json:"name,omitempty" bson:"name,omitempty"`
+	DataType     DataType  `json:"dataType,omitempty" bson:"dataType,omitempty"`
```

Comment:
what's the limit on the cache and what's the usage for. DataType ?

Reply:
T can be anything 
datatype s a string that helps us filter according to T type